packages/core/content-manager/server/src/history/services/history.ts

```diff
@@ -121,7 +187,7 @@ const createHistoryService = ({ strapi }: { strapi: LoadedStrapi }) => {
             this.createVersion({
               contentType: contentTypeUid,
               data: omit(fieldsToIgnore, document),
-              schema: omit(fieldsToIgnore, strapi.contentType(contentTypeUid).attributes),
+              schema: omit(fieldsToIgnore, strapi.getModel(contentTypeUid).attributes),
```

Comment:
could you just omit non visible attributes here?

Reply:
I there there are some we would want to keep. Even if a field is not visible in the edit view, its value would still be restored when restoring an older history version. So we'll display them on the history page regardless of whether they're hidden, so that users know what they're restoring

Reply:
ok! just in case, that means this would be restoring review workflow fields, creator fields, publication state fields.

Also, what happens if a user enables DP on a content type? Does the history resets?

Reply:
Ah so when you say non-visible you mean those, not the CTB 'visible' setting. I that case I can do that. 

Regarding enabling D&P, I think it can be a problem for later. There was a similar issue on content releases, we could use their solution

Reply:
>Ah so when you say non-visible you mean those, not the CTB 'visible' setting.
I do think the CTB 'visible' setting is the same setting, back in the day there was a bug that if you made a field not visible in the CTB , it disapeared from the CTB list too 😆 , dont know if that has been fixed.

>Regarding enabling D&P, I think it can be a problem for later. There was a similar issue on content releases, we could use their solution
Makes sense! Probably could be somethign to handle when disabling locales too, to make sure the history of the locale is disabled.

Reply:
in that case I think it's okay to keep our [list of fields to omit](https://github.com/strapi/strapi/blob/f32aed06cb5301f34314d71e1e9fbc8649f46638/packages/core/content-manager/server/src/history/constants.ts#L2-L12), since we don't want to skip any other ones 🤔

Reply:
I would try to avoid such lists generally, as it makes it complicated to maintain them when we add new fields.